src/index.js

```diff
@@ -69,7 +69,15 @@ function preInit(initPromise) {
     dataView.preInit(initPromise);
 }
 
-function init(promise, resolve, reject) {
+function init() {
+    // setup init promise
+    let resolve;
+    let reject;
```

Comment:
You can stick the `onTimeout` (and `setTimeout`) fn/calls inside the promise to avoid having to declare these variables.